tools/dev.js

```diff
@@ -0,0 +1,87 @@
+const path = require("path");
+const { createServer } = require("http");
+const chokidar = require("chokidar");
+const karma = require("karma");
+const serve = require("serve-handler");
+const colors = require("colors");
+const serveConfig = require("../serve.json");
+const { Builder } = require("./builder");
+
+const KARMA_PORT = 9876;
+const SERVE_PORT = 5000;
+
+const karmaConfig = karma.config.parseConfig(
+  path.join(__dirname, "../karma.conf.js"),
+  {
+    browsers: ["ChromeHeadless"],
+    autoWatch: false,
+    port: KARMA_PORT,
+    logLevel: karma.constants.LOG_WARN,
+  }
+);
+const karmaServer = new karma.Server(karmaConfig);
+
+main();
```

Comment:
Main looks kinda redundant, tbh... maybe we should just ditch main() and just inline all this?